libraries/rush-sdk/src/proxy.ts

```diff
@@ -0,0 +1,288 @@
+   */
+  rushJsonSearchFolder?: string;
+
+  /**
+   * A cancellation token that the caller can use to prematurely abort the operation.
+   */
+  abortSignal?: AbortSignal;
+
+  /**
+   * Allows the caller to monitor the progress of the operation.
+   */
+  onNotifyEvent?: SdkNotifyEventCallback;
+}
+
+/**
+ * Exposes operations that control how the `@microsoft/rush-lib` engine is
+ * located and loaded.
+ * @public
+ */
+// eslint-disable-next-line @typescript-eslint/no-namespace
+export namespace rushSdkProxy {
+  function _checkForCancel(
+    abortSignal: AbortSignal,
+    onNotifyEvent: SdkNotifyEventCallback | undefined,
+    progressBarPercent: number | undefined
+  ): boolean {
+    if (!abortSignal.aborted) {
+      return true;
+    }
+
+    if (onNotifyEvent) {
+      onNotifyEvent({
+        state: 'aborted',
+        logMessage: {
+          messageType: 'info',
+          message: `The operation was cancelled`
+        },
+        progressBarPercent
+      });
+    }
+
+    return false;
+  }
+
+  /**
+   * Manually load the Rush engine based on rush.json found for `rushJsonSearchFolder`.
+   * This API supports an optional cancellation token and callback for status updates
+   * during the operation.
+   */
+  export async function loadAsync(options?: ILoadSdkAsyncOptions): Promise<void> {
+    // SCENARIO 5: The rush-lib engine is loaded manually using rushSdkProxy.loadAsync().
+
+    if (!options) {
+      options = {};
+    }
+
+    const onNotifyEvent: SdkNotifyEventCallback | undefined = options.onNotifyEvent;
+    let progressBarPercent: number | undefined = undefined;
+
+    const abortSignal: AbortSignal = options.abortSignal ?? { aborted: false };
+
+    try {
+      const rushJsonSearchFolder: string = options.rushJsonSearchFolder ?? process.cwd();
+      const rushJsonPath: string | undefined = tryFindRushJsonLocation(rushJsonSearchFolder);
+      if (!rushJsonPath) {
+        throw new Error(
+          'Unable to find rush.json in the current folder or its parent folders.\n' +
+            'This tool is meant to be invoked from a working directory inside a Rush repository.'
+        );
+      }
+      const monorepoRoot: string = path.dirname(rushJsonPath);
+
+      const rushJson: JsonObject = JsonFile.load(rushJsonPath);
+      const { rushVersion } = rushJson;
+
+      const installRunNodeModuleFolder: string = path.join(
+        monorepoRoot,
+        `common/temp/install-run/@microsoft+rush@${rushVersion}`
+      );
+
+      try {
+        // First, try to load the version of "rush-lib" that was installed by install-run-rush.js
+        if (onNotifyEvent) {
+          onNotifyEvent({
+            state: 'running',
+            logMessage: {
+              messageType: 'info',
+              message: `Trying to load  ${RUSH_LIB_NAME} installed by install-run-rush`
+            },
+            progressBarPercent
+          });
+        }
```

Comment:
```suggestion
        onNotifyEvent?.({
          state: 'running',
          logMessage: {
            messageType: 'info',
            message: `Trying to load  ${RUSH_LIB_NAME} installed by install-run-rush`
          },
          progressBarPercent
        });
```

Reply:
Though would probably be good to wrap this in a helper, since this is mostly boilerplate.

Reply:
The syntax is more concise now that `state` has been removed